torch/nn/parallel/distributed.py

```diff
@@ -692,6 +694,24 @@ def __init__(
             self._set_static_graph()
 
         self._setup_in_backward_optimizers()
+        self._register_load_state_dict_pre_hook(
+            self._pre_load_state_dict_hook, with_module=True
+        )
+
+    @staticmethod
+    def _pre_load_state_dict_hook(
+        module: nn.Module,
+        state_dict: Dict[str, Any],
+        prefix: str,
+        *args: Any,
+    ) -> None:
+        """
+        ``_pre_state_dict_hook` is called before ``self._load_from_state_dict()``
+        is called. For ``DistributedDataParallel``, it will add back the module
+        prefix so that non-DDP modules (i.e. local nn.Modules) can be loaded into
+        DDP module properly.
+        """
+        _replace_by_prefix(state_dict, prefix, prefix + f"{_DDP_PREFIX}.")
```

Comment:
where is this implemented?

Reply:
sorry need to import, but it is implemented in torch/distributed/utils.